src/UI/Component/Modal/Interruptive.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component\Button;
+
+/**
+ * Interface Interruptive
+ *
+ * @package ILIAS\UI\Component\Modal
+ */
+interface Interruptive extends Modal
+{
+
+    /**
+     * Get the modal with the given button as action
+     *
+     * @param Button\Button $button
+     * @return Interruptive
+     */
+    public function withActionButton(Button\Button $button);
```

Comment:
The question is, if the buttons may be part of another included UI component, e.g. a form.